client/components/close-on-escape/test/index.jsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import React from 'react';
+import ReactDom, { unmountComponentAtNode } from 'react-dom';
+import { spy } from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import useFakeDom from 'test/helpers/use-fake-dom';
+import CloseOnEscape from '../';
+
+const simulateEscapeKeydown = () => (
+	document.dispatchEvent( new window.KeyboardEvent( 'keydown', { keyCode: 27 } ) )
+);
+
+describe( 'CloseOnEscape', () => {
+	describe( 'rendering', () => {
+		it( 'renders nothing', () => {
+			const wrapper = shallow( <CloseOnEscape /> );
+			expect( wrapper.type() ).to.be.a( 'null' );
+		} );
+	} );
+
+	describe( 'escape keydown event', () => {
```

Comment:
Hi @gwwar!
If you get some free time could you take a look over this test please?
As I was writing it I felt quite torn... On one hand this feels like overkill, on the other hand I feel that it's important to test the queuing and dequeuing (in the correct order) of the events and I couldn't think of a more concise way...

No rush though as I've still got a good chunk of work to do before I'd consider this to be mergable! :D